tests/integration/data/kfp_1.7_stable_generic_install.yaml.j2

```diff
@@ -0,0 +1,198 @@
+bundle: kubernetes
+applications:
+  argo-controller:
+    charm: argo-controller
+    channel: 3.3/stable
+    revision: 236
+    series: focal
+    resources:
+      oci-image: 185
+    scale: 1
+    constraints: arch=amd64
+  argo-server:
+    charm: argo-server
+    channel: 3.3/stable
+    revision: 185
+    series: focal
+    resources:
+      oci-image: 186
+    scale: 1
+    constraints: arch=amd64
+  dex-auth:
+    charm: dex-auth
+    channel: 2.31/stable
+    revision: 224
+    series: focal
+    resources:
+      oci-image: 318
+    scale: 1
+    options:
+      public-url: http://10.64.140.43.nip.io
+      static-password: admin
+      static-username: admin
+    constraints: arch=amd64
+    trust: true
+  istio-ingressgateway:
+    charm: istio-gateway
+    channel: 1.16/stable
+    revision: 551
+    series: focal
+    scale: 1
+    options:
+      kind: ingress
+    constraints: arch=amd64
+    trust: true
+  istio-pilot:
+    charm: istio-pilot
+    channel: 1.16/stable
+    revision: 551
+    series: focal
+    scale: 1
+    options:
+      default-gateway: kubeflow-gateway
+    constraints: arch=amd64
+    trust: true
+  kfp-db:
+    charm: mysql-k8s
+    channel: 8.0/edge
+    revision: 85
+    series: jammy
+    resources:
+      mysql-image: 84
+    scale: 1
+    constraints: arch=amd64 mem=600M
+    storage:
+      database: kubernetes,1,1024M
+    trust: true
+  minio:
+    charm: minio
+    channel: ckf-1.7/stable
+    revision: 186
+    series: focal
+    resources:
+      oci-image: 468
+    scale: 1
+    constraints: arch=amd64
+    storage:
+      minio-data: kubernetes,1,10240M
+  oidc-gatekeeper:
+    charm: oidc-gatekeeper
+    channel: ckf-1.7/stable
+    revision: 176
+    series: focal
+    resources:
+      oci-image: 308
+    scale: 1
+    options:
+      public-url: http://10.64.140.43.nip.io
+    constraints: arch=amd64
+{%- if local_build == false %}
```

Comment:
Yeah, the case that you mentioned is not covered here. Another thing I've seen in other charms ([cos-lite](https://github.com/canonical/cos-lite-bundle/blob/main/bundle.yaml.j2)) is that they have a condition per charm, so they will render only iff the context contains a local charm path, otherwise, they will use a charm from CH.
I did not want to do that because it would've been a lot more configuration on the test side (helpers). If you think it is necessary, though, I can work on it a bit more.

Reply:
I can understand this flexibility wasn't part of the initial scope, but since this refactor is moving some existing integration tests away from the charm and to the bundle I think we're harming the developer experience unless single-charm building is enabled.  Looking at the PRs to charms in this repo they're usually edits to a single charm not the entire bundle, so the typical development experience is focused on editing one charm in relation to others that are unchanged.  Since building the whole bundle locally takes ~1hr (I might be misremembering, but at least 30 min) the development experience after this PR merges sounds frustrating imo

Reply:
I think the developer experience hasn't really changed between the current state of the repo (building and deploying ALL charms) and the proposal in this PR, but let me know if it has. 
Now that of it, the kind of optimisation you mention (changes in a single charm would trigger builds only for that charm) belong to a different effort, which we can combine with cached builds, and really improve the CI times. For now, I think it is better keep the scope of this PR as is.

Reply:
From the spec, I thought the plan was removing kfp-api's charm integration tests and instead using the bundle tests.  I don't see that change in this PR, but I'm guessing from the spec that it is planned.  Is that the case?  Using these bundle tests as kfp-api's integration test is the regression I'm thinking of

Reply:
No, those will remain there.